apps/test/unit/aiDifferentiation/AiDiffChatTest.jsx

```
@@ -497,6 +497,112 @@ describe('AiDiffChat', () => {
+    fireEvent.click(submit_btn);
+
+    //After click, but before server response, user message editor should be disabled
+    expect(submit_btn).not.toBeEnabled();
+    expect(textbox).not.toBeEnabled();
+
+    const responseEventData = {
+      chatContext: {
+        type: AiDiffContext.LESSON,
+        lessonId: 2,
+      },
+      scriptName: 'test_lesson',
+      role: Role.USER,
+      isPreset: false,
+      text: userMessage,
+      threadId: 3,
+      url: window.location.href,
+    };
+    const responseEventData2 = {
+      chatContext: {
+        type: AiDiffContext.LESSON,
+        lessonId: 2,
+      },
+      scriptName: 'test_lesson',
+      role: Role.ASSISTANT,
+      isPreset: false,
+      text: "Beep boop I'm a bot",
+      threadId: 3,
+      url: window.location.href,
+    };
+
+    //sends the api call then logs the user message and the bot message
+    await waitFor(() => {
+      expect(fetchStub).toHaveBeenCalledWith(
+        '/aidiff_threads/3/chat_completion',
+        JSON.stringify({
+          inputText: responseEventData.text,
+          isPreset: false,
+          presetChipText: null,
+        }),
+        true,
+        {
+          'Content-Type': 'application/json',
+        }
+      );
+      expect(sendEventSpy).toHaveBeenCalledWith(
+        EVENTS.AI_DIFF_CHAT_EVENT,
+        responseEventData,
+        PLATFORMS.STATSIG
+      );
+      expect(sendEventSpy).toHaveBeenCalledWith(
+        EVENTS.AI_DIFF_CHAT_EVENT,
+        responseEventData2,
+        PLATFORMS.STATSIG
+      );
+    });
+    //two user message
+    expect(
+      screen.getAllByLabelText(i18n.aiChatMessageUser())[1]
+    ).toHaveTextContent(userMessage);
+    //second bot message has the response
+    expect(
+      screen.getAllByLabelText(i18n.aiChatMessageBot())[1]
+    ).toHaveTextContent("Beep boop I'm a bot");
+
+    //User message editor should be enabled once we have a server response
+    expect(submit_btn).not.toBeEnabled();
```

The  comment and the expect statement seem backwards to each other?

---

In general I don't like adding these kinds of comments because they can hide errors, be wrong or out of date.

---

turns out "not" is a critical word to leave out 😬